File: 2021/56. Merge Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());        
        vector<vector<int>> res = {intervals[0]};
        for (int i = 1; i < intervals.size(); i++) {
            if (intervals[i][0] <= res.back()[1]) {
                res.back()[1] = max(intervals[i][1], res.back()[1]);
            } else {
                res.push_back(intervals[i]);
            }
        }
        return res;
    }
};

File: 2024/56. Merge Intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        def sortFunc(interval: List[int]):
            return interval[0]
        intervals.sort(key=sortFunc)
        i = 0
        ans = list()
        while i < len(intervals):
            curr = intervals[i]
            while i < len(intervals) and intervals[i][0] <= curr[1]:
                curr[1] = max(intervals[i][1], curr[1])
                i += 1
            ans.append(curr)
        return ans
